src/mysqlsh_helpers.py

```diff
@@ -56,19 +73,45 @@ def configure_mysql_users(self):
         """
         pass
 
-    def configure_instance(self):
+    def configure_instance(self) -> None:
         """Configure the instance to be used in an InnoDB cluster."""
-        pass
+        _script = f"""
+dba.configure_instance('{self.cluster_admin_user}:{self.cluster_admin_password}@{self.instance_address}')
+my_shell = shell.connect('{self.cluster_admin_user}:{self.cluster_admin_password}@{self.instance_address}')
+my_shell.run_sql("RESTART;");
+        """
 
-    def create_cluster(self):
+        try:
+            logger.debug("Configuring instance for InnoDB")
+            self._run_mysqlsh_script(_script)
+
+            logger.debug("Waiting until MySQL is restarted")
+            self._wait_until_mysql_connection()
+        except subprocess.CalledProcessError as e:
+            logger.exception(f"Failed to configure instance: {self.instance_address}", exc_info=e)
+            raise MySQLInstanceConfigureError(e.stderr)
+
+    def create_cluster(self) -> None:
         """Create an InnoDB cluster with Group Replication enabled."""
         pass
 
-    def add_instance_to_cluster(self):
+    def add_instance_to_cluster(self) -> None:
         """Add an instance to the InnoDB cluster."""
         pass
 
-    def run_mysqlsh_script(self, script: str) -> AnyStr:
+    @retry(reraise=True, stop=stop_after_delay(30), wait=wait_fixed(5))
+    def _wait_until_mysql_connection(self) -> None:
```

Comment:
Shouldn't the return type be `AnyStr`? Otherwise, you can remove the return statement at the end of this function.

Reply:
good point, removed return type in 46822799e0a54478e31361bb4cbea6fd70c4e454. we will add it back if we find a need for it in the future